core/src/test/java/org/bitcoinj/crypto/HDPathTest.java

```diff
@@ -31,7 +33,7 @@
     public void testPrimaryConstructor() throws Exception {
         HDPath path = new HDPath(true, Collections.<ChildNumber>emptyList());
         Assert.assertTrue("Has private key returns false incorrectly", path.hasPrivateKey);
-        Assert.assertEquals("Path not empty", path.size(), 0);
```

Comment:
Fixing the use of `assertEquals()` in the entire project would be a nice separate PR.